### ISSUE TITLE
Substitution Problems

### ISSUE THREAD

**Issue by marcantonio** [gongo/json-reformat#17]
I noticed with recent versions the following 2 undesirable behaviors:
`"foo": {},`

is changed to:
`"foo": null,`

and
`"conf": "^(amq\\.gen.*|amq\\.default)$",`

is changed to:
`"conf": "^(amq\.gen.*|amq\.default)$",`

Note, I have `(setq json-reformat:pretty-string? t)` in my config to prevent forward slashes from being escaped.

**Comment by gongo** [comment]
Thanks for issue. I'll try :pizza:

**Comment by gongo** [comment]
## 1st

> "foo": {}, is changed to: "foo": null,

Sorry, it is difficult to solve this problem for me.

### Because

`json-reformat` parse json string using `json-read-from-string` function that built-in emacs.
Empty-hash and null, `json-read-from-string` recognized as `nothing`.

```lisp
(json-read-from-string "{\"foo\": {}}")
;; ((foo))
(json-read-from-string "{\"foo\": null}")
;; ((foo))
```

So, `json-reformat` output `null`.

## 2nd

I resolve ! GH-18

**Comment by gongo** [comment]
@marcantonio 

2nd problem is solved at `json-reformat-20150310.555`. Please try !

**Comment by gongo** [comment]
A week has passed. I determined that it has completed :pizza:
